app/main.py

```
@@ -87,11 +89,34 @@ def setup_phoenix_tracing(self):
+        frontend_url = os.getenv("NEXT_PUBLIC_BASE_URL") or os.getenv("FRONTEND_URL")
+        if frontend_url:
+            # Extract base URL without path
+            from urllib.parse import urlparse
+
+            parsed = urlparse(frontend_url)
+            base_url = f"{parsed.scheme}://{parsed.netloc}"
+            if base_url not in origins:
+                origins.append(base_url)
+
+        # Always use specific origins with credentials for proper CORS support
+        logging.info(f"CORS configured with origins: {origins}")
```

Ruff (0.14.8)

`logging` may be undefined, or defined from star imports

(F405)